### ISSUE TITLE
Windows Server Core Black Screen

### ISSUE THREAD

**Issue by JSuenram** [Ylianst/MeshCentral#3454]
Hi,
while connecting to GUI-Installed Windows Servers is working flawless, servers with Core-Install mostly show only black screen.
You can move mouse and send keyboard strokes, but nothing is shown. However viewing same system in VM-Console shows indeed that MC is connected and data is send. This problem is mostly on Server 2022 exclusive so far....

**Comment by krayon007** [comment]
Doesn't Windows Server Core not have a desktop UX?

**Comment by Ylianst** [comment]
I have never installed Windows Server Core myself, but [according to this](https://docs.microsoft.com/en-us/windows-server/administration/server-core/what-is-server-core):

```
...In other words, there is no desktop in Server Core, by design. 
```

So right, is that not normal? We should detect this version and remove the remove desktop tab?

**Comment by JSuenram** [comment]
> I have never installed Windows Server Core myself, but [according to this](https://docs.microsoft.com/en-us/windows-server/administration/server-core/what-is-server-core):
> 
> ```
> ...In other words, there is no desktop in Server Core, by design. 
> ```
> 
> So right, is that not normal? We should detect this version and remove the remove desktop tab?

Wrong... Server Core has a GUI... you can login with RDP, use CMD, PowerShell TaskManager and even install FOD to have EventViewer and other tools. You can even run a Webbrowser. So remote-control of a Core-Server is absolutely daily business for Admins.... So this is needed and wanted!

**Comment by krayon007** [comment]
I'll have to do some investigation. I setup Windows Core 2022.. It looks like desktop remoting actually works, as it shows the privacy bar, and it shows all the spawned windows/applications. It looks like it only doesn't show the initial cmd console. The other issue, is that if the screen times out, you have to hit ctrl-alt-del to bring the console back, but we currently use SAS to inject ctrl-alt-del, but SAS doesn't exist on Server Core.

**Comment by OutbackMatt** [comment]
There are registry settings used to start a command prompt (and other items)
I use this to start Corefig, or explorer alternatives or task manager  etc on my HyperV Core servers

Should be the same

HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Run

Perhaps the server doesn't see that meshcentral rdp is a new session

**Comment by JSuenram** [comment]
Any update on this?

**Comment by krayon007** [comment]
So the main issue with this, is that it looks like core, doesn't instantiate a default desktop session. Or at least, the initial cmd window is not rendered in the default session. However, a simple workaround could be to use the new webrdp functionality that was recently added. That will actually create a new login session.
